vault/overlays/nerc-ocp-infra/backup-job/task.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: backup-vault
+spec:
+  workspaces:
+    - name: snapshots
+      mountPath: /snapshots
+    - name: backup-job
+      mountPath: /backup-job
+  steps:
+    - name: create-snapshot
+      image: quay.io/operate-first/opf-toolbox:v0.8.0
+      workingDir: /workspace
+      script: |
+        #!/bin/sh
+        set -e
+
+        echo 'Logging into nerc-vault-0 to retrieve leader'
+        oc exec -i nerc-vault-0 -- vault operator raft list-peers --format=json > /tmp/peers.json
+        leader=$(yq e '.data.config.servers.[] | select(.leader == "true") | .address' -P /tmp/peers.json | cut -d '.' -f1)
+
+        echo retrieved leader: ${leader}
+        echo Creating Snapshot in leader pod ${leader}...
+
+        export snapshot_file_path=$(echo snapshot-"`date +"%F_T%H-%M-%S"`".snap)
+        oc exec -i $leader -- vault operator raft snapshot save /tmp/backup_job_snapshot.snap
+
+        echo Snapshot created in Vault Leader pod. Moving snapshot from pod to backup PVC.
+
+        snapshot_pvc_path=$(workspaces.snapshots.path)
+        oc cp $(context.taskRun.namespace)/$leader:/tmp/backup_job_snapshot.snap ${snapshot_pvc_path}/${snapshot_file_path}
+        oc exec -i $leader -- rm /tmp/backup_job_snapshot.snap
+
+        echo Backup Snapshot saved at ${snapshot_pvc_path}/$snapshot_file_path
+        echo Listing current snapshots in backedup PVC
+
+        echo --------------------------------------------
+        ls -lha ${snapshot_pvc_path}
+        echo --------------------------------------------
+
+        echo "Done!"
+    - name: backup-snapshot-to-s3
+      image: quay.io/operate-first/mc:RELEASE.2022-06-17T02-52-50Z
+      workingDir: /workspace
+      script: |
+        #!/bin/sh
+        set -e
+
+        S3EndPoint=$(cat /backup-job/S3EndPoint)
+        S3AccessKey=$(cat /backup-job/S3AccessKey)
+        S3SecretKey=$(cat /backup-job/S3SecretKey)
+        S3Bucket=$(cat /backup-job/S3BucketName)
+        S3Retention=$(cat /backup-job/S3Retention)
+        PVCRetention=$(cat /backup-job/PVCRetention)
```

Comment:
If we used the secret in an `envFrom` block (either on this step or in a `stepTemplate`), we would already have these values as variables and wouldn't need this block at all.

(This is true for anywhere you see `somevar=$(cat /backup-job/something)`)